backend/de.metas.edi/src/main/java/de/metas/edi/process/EDI_GenerateCSV_FileForSSCC_Labels.java

```diff
@@ -31,30 +35,69 @@
 import de.metas.util.Services;
 import lombok.NonNull;
 import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.dao.IQueryFilter;
 import org.compiere.SpringContextHolder;
+import org.compiere.model.I_AD_Zebra_Config;
+import org.compiere.model.I_C_BP_PrintFormat;
+import org.compiere.model.I_C_BPartner;
 
 import java.util.Collection;
+import java.util.List;
 
 public abstract class EDI_GenerateCSV_FileForSSCC_Labels extends JavaProcess implements IProcessPrecondition
 {
 	protected final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private static final AdMessageKey MSG_DIFFERENT_ZEBRA_CONFIG_NOT_SUPPORTED = AdMessageKey.of("WEBUI_ZebraConfigError");
 
 	@Override public ProcessPreconditionsResolution checkPreconditionsApplicable(@NonNull IProcessPreconditionsContext context)
 	{
 		if (context.getSelectionSize().isNoSelection())
 		{
 			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
 		}
+		else  if (context.getSelectionSize().isMoreThanOneSelected())
+		{
+			final IQueryFilter<I_EDI_Desadv> selectedRecordsFilter = context.getQueryFilter(I_EDI_Desadv.class);
+			final List<Integer> formatLinesWithNoZebraConfig = queryBL
+					.createQueryBuilder(I_EDI_Desadv.class)
+					.filter(selectedRecordsFilter)
+					.andCollect(I_EDI_Desadv.COLUMNNAME_C_BPartner_ID, I_C_BPartner.class)
+					.andCollectChildren(I_C_BP_PrintFormat.COLUMNNAME_C_BPartner_ID, I_C_BP_PrintFormat.class)
+					.addEqualsFilter(I_C_BP_PrintFormat.COLUMN_AD_Zebra_Config_ID, null)
+					.create()
+					.listDistinct(I_C_BP_PrintFormat.COLUMNNAME_C_BP_PrintFormat_ID, Integer.class);
+
+
+			final List<Integer> zebraConfigIds = queryBL
+					.createQueryBuilder(I_EDI_Desadv.class)
+					.filter(selectedRecordsFilter)
+					.andCollect(I_EDI_Desadv.COLUMNNAME_C_BPartner_ID, I_C_BPartner.class)
+					.andCollectChildren(I_C_BP_PrintFormat.COLUMNNAME_C_BPartner_ID, I_C_BP_PrintFormat.class)
+					.andCollect(I_C_BP_PrintFormat.COLUMNNAME_AD_Zebra_Config_ID, I_AD_Zebra_Config.class)
+					.create()
+					.listDistinct(I_AD_Zebra_Config.COLUMNNAME_AD_Zebra_Config_ID, Integer.class);
+
+			if (formatLinesWithNoZebraConfig.size() > 0 && zebraConfigIds.size() > 0)
```

Comment:
Not sure I understand the purpose of `formatLinesWithNoZebraConfig`..i might guess the idea..but then, what about bpartners that have not even a `C_BP_PrintFormat`

Reply:
Idea: 
* Add a new method ``retrieveZebraConfigId`` in IBPartnerDAO. Will return the ID of the zebra config from the bp print format if it exists and default otherwise.
* modify the precondition like this: 

```java
	queryBL
				.createQueryBuilder(I_EDI_Desadv.class)
				.filter(queryFilter)
				.create()
				.list()
				.stream()
				.map(ediDesadv -> BPartnerId.ofRepoId(ediDesadv.getC_BPartner_ID()))
				.map(bpartnerId -> bpartnerDAO.retrieveZebraConfigId(bpartnerId))
				.collect(Collectors.toSet())
				.size();
```

=> size must be 1 for acceptance.

Reply:
I added the method `retrieveZebraConfigId  `in `ZebraConfigRepository` I think the logic belongs to the zebraConfig and should be there

Reply:
You are right.